### ISSUE TITLE
Compile errors with TypeScript 2.4.1

### ISSUE THREAD

**Issue by dasa** [dojo/typings#126]
These are the errors I'm getting with tsc on the master branch:

```console
$ tsc index.d.ts 
dijit/1.11/form.d.ts(1589,13): error TS2430: Interface 'NumberSpinner' incorrectly extends interface '_Spinner'.
  Types of property 'filter' are incompatible.
    Type '(value: number) => number' is not assignable to type '<T>(val: T) => T'.
      Types of parameters 'value' and 'val' are incompatible.
        Type 'T' is not assignable to type 'number'.
dijit/1.11/form.d.ts(1627,13): error TS2430: Interface 'NumberTextBox' incorrectly extends interface 'RangeBoundTextBox'.
  Types of property 'filter' are incompatible.
    Type '(value: number) => number' is not assignable to type '<T>(val: T) => T'.
      Types of parameters 'value' and 'val' are incompatible.
        Type 'T' is not assignable to type 'number'.
dijit/1.11/form.d.ts(1686,13): error TS2430: Interface 'SimpleTextarea' incorrectly extends interface 'TextBox'.
  Types of property 'filter' are incompatible.
    Type '(value: string) => string' is not assignable to type '<T>(val: T) => T'.
      Types of parameters 'value' and 'val' are incompatible.
        Type 'T' is not assignable to type 'string'.
dijit/1.11/layout.d.ts(89,13): error TS2430: Interface 'LayoutContainer' incorrectly extends interface '_LayoutWidget'.
  Types of property 'removeChild' are incompatible.
    Type '<T extends _WidgetBase>(child: T) => void' is not assignable to type '<T extends _WidgetBase>(widget: number | T) => void'.
      Types of parameters 'child' and 'widget' are incompatible.
        Type 'number | T' is not assignable to type '_WidgetBase'.
          Type 'number' is not assignable to type '_WidgetBase'.
```

**Comment by dasa** [comment]
Workaround: `tsc --noStrictGenericChecks`

**Comment by dasa** [comment]
@dylans - now that this was fixed by PR #127, it'd be great to have a new release pushed to npm. Thanks.

**Comment by dylans** [comment]
@dasa, of course, here you go: https://github.com/dojo/typings/releases/tag/1.11.7
